front/components/assistant_builder/AssistantBuilder.tsx

```diff
@@ -255,15 +257,24 @@ export default function AssistantBuilder({
           modelConfig.contextSize * 0.9
         }`
       );
-      setInstructionsError(`Instructions may exceed context size window.`);
-      valid = false;
-    } else {
-      setInstructionsError(null);
+      localInstructionError = `Instructions may exceed context size window.`;
     }
 
-    if (!builderState.actions.every((a) => hasActionError(a) === null)) {
-      valid = false;
-    }
+    // We only keep the last error. If there are multiple errors, the user will have to fix them one by one.
+    setInstructionsError(localInstructionError);
+
+    // Check if there are any errors in the actions
+    const actionErrors = builderState.actions
+      .map((a) => hasActionError(a))
+      .filter(Boolean) as string[];
+
+    setActionsErrors(actionErrors);
+
+    const valid =
+      actionErrors.length === 0 &&
```

Comment:
Why passing through a local state in the function? (i.e. Instead of localDescriptionError === null why not descriptionError === null? - same for instruction & handle)

Reply:
This one is for a reason, or at least I think so :)
setXXX is not instant and also, if I used the global state var, then i have to add it to the dependencies.